src/pages/PlanPage/subRoutes/UpgradePlanPage/UpgradeForm/ProPlanControls/ProPlanControls.spec.jsx

```diff
@@ -0,0 +1,915 @@
+        expect(organizationTitle).toBeInTheDocument()
+      })
+
+      it('renders the owner', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const ownerTitle = await screen.findByText(/codecov/)
+        expect(ownerTitle).toBeInTheDocument()
+      })
+
+      it('renders the billing controls', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const billingControls = await screen.findByText(/Billing Controls/)
+        expect(billingControls).toBeInTheDocument()
+      })
+
+      it('renders minimum seat number of 2 for basic plan', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const minimumSeat = await screen.findByRole('spinbutton')
+        expect(minimumSeat).toHaveValue(2)
+      })
+
+      it('renders validation error when the user selects less than 2 seats', async () => {
+        const { user } = setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const input = await screen.findByRole('spinbutton')
+        await user.type(input, '{backspace}{backspace}{backspace}')
+        await user.type(input, '1')
+
+        const proceedToCheckoutButton = await screen.findByRole('button', {
+          name: /Proceed to Checkout/,
+        })
+        expect(proceedToCheckoutButton).toBeDisabled()
+
+        const error = screen.getByText(
+          /You cannot purchase a per user plan for less than 2 users/
+        )
+        expect(error).toBeInTheDocument()
+      })
+
+      it('renders the user count component', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const userCount = await screen.findByText('User Count')
+        expect(userCount).toBeInTheDocument()
+      })
+
+      it('renders the total banner component', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const totalBanner = await screen.findByText('Total Banner')
+        expect(totalBanner).toBeInTheDocument()
+      })
+
+      it('renders the proceed to checkout for the update button', async () => {
+        setup({ planValue: Plans.USERS_BASIC })
+        render(<ProPlanControls />, { wrapper: wrapper() })
+
+        const proceedToCheckoutButton = await screen.findByRole('button', {
+          name: /Proceed to Checkout/,
+        })
+        expect(proceedToCheckoutButton).toBeInTheDocument()
+      })
+
+      describe('when the mutation is successful', () => {
+        // This test works because users-pr-inappy is the default choice, but by me mocking the billingControls
+        // I lose the ability to change that and test users-basic > monthly, only users-basic > yearly. Wdyt?
```

Comment:
I would suggest not mocking this out, so we can get the whole flow going in a more integrated test manner